astrapy/api.py

```diff
@@ -0,0 +1,150 @@
+import logging
+import httpx
+from typing import Any, Dict, Optional, TypeVar, cast
+
+from astrapy.types import API_RESPONSE
+from astrapy.utils import amake_request, make_request
+
+T = TypeVar("T", bound="APIRequestHandler")
+AT = TypeVar("AT", bound="AsyncAPIRequestHandler")
+
+
+logger = logging.getLogger(__name__)
+
+
+class APIRequestError(ValueError):
+    def __init__(self, response: httpx.Response) -> None:
+        super().__init__(response.text)
+
+        self.response = response
+
+    def __repr__(self) -> str:
+        return f"{self.response}"
+
+
+class APIRequestHandler:
```

Comment:
I'm not sure to see the interest of the `APIRequestHandler ` compared to static utils methods.